serving/src/main/java/feast/serving/service/OnlineServingServiceV2.java

```diff
@@ -102,14 +103,31 @@ public GetOnlineFeaturesResponse getOnlineFeatures(GetOnlineFeaturesRequestV2 re
 
             FeatureTableSpec featureTableSpec =
                 specService.getFeatureTableSpec(projectName, feature.get().getFeatureReference());
+            FeatureProto.FeatureSpecV2 tempFeatureSpecV2 =
```

Comment:
Can we build some map outside of the loop to avoid this filtering every iteration?